file_loader_tool.py

```
@@ -1,162 +1,335 @@
+                        msg = f"Skipped (binary) {file_path}"
+                        self.skipped_files.append(msg)
+                        self._log(msg, level="WARNING")
+                        processed_count += 1
+                        if progress_callback is not None:
+                            progress_callback('file_loader', processed_count, total_estimate, str(root_path))
+                        continue
+                    # Attempt to read using encoding fallback strategy
+                    content, used, err = self._read_text_with_fallback(file_path)
+                    if content is None:
+                        # Propagate meaningful failure preserving context
+                        raise RuntimeError(f"Failed to decode text file: {file_path}") from err
```

Ruff (0.14.4)

Avoid specifying long messages outside the exception class

(TRY003)